modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js

```diff
@@ -0,0 +1,282 @@
+import getCN from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {DragDropContext as dragDropContext} from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+import CriteriaSidebar from '../criteria_sidebar/CriteriaSidebar.es';
+import {sub} from '../../utils/utils.es';
+import CriteriaBuilder from './CriteriaBuilder.es';
+import {buildQueryString, translateQueryToCriteria} from '../../utils/odata.es';
+import Conjunction from './Conjunction.es';
+
+class ContributorsBuilderComp extends React.Component {
+	constructor(props) {
+		super(props);
+		this.classes = getCN(
```

Comment:
Also, why are we setting the classes here? This should be in the render function.

Reply:
I was setting the classes there so they don't get generated on every render. They don't depend on  props or state, so it make sense to do it just on the instantiation of the component. Does that make any sense?

I'm not positive on any performance gains. It's just an intuition: keep the render light.

Reply:
If it doesn't depend on props or state, we can just set the classes as a static string on the jsx element itself. Thankfully react is smart enough that it does actually "render" every element every time the method is called, it is actually just "reconciling" the virtual dom. Since the classes never change, it won't have to re-render those ever, so  I don't think this will ever be a performance issue.

If we do use "classnames" are trying to avoid calling "getCN" in render, I wouldn't worry about that too much. The performance of that is negligible and its more important that the classNames are easily read within the render method.